### DISCUSSION TITLE
Nothing happens when I trigger my extension

### DISCUSSION THREAD

**Discussion by Drisate** [microsoft/vscode-discussions#327] [Extension Development QnA]
Hey guys!

I am playing with the OpenAi API and trying to create a VSC extension for my personal use to both understand the VSC extension principals but also to see what else I can build using OpenAI

The premise of my extension is very simple:

1. The user types a description of what he intends to code:
Ex: Create a laravel model for table customer using column names: name, address, phone. 
2. After that, the user selects the text, right clicks and selects "OpenAi Generate" from the right click menu option.
3. I then use a showQuickPick() window so the user can select in what language to generate the code in
Ex: 'python', 'javascript', 'csharp', 'java', 'go', 'ruby', 'swift', 'php', 'powershell'
In this case, the user would select PHP
4. That request is sent to OpenAi and the response replaces the selected text in the window.

Eventually, I would like OpenAi to read the whole document, find all annotations of 'TO-DO' and complete the code

To make the OpenAi Generate option, I added this in my package.json

```
    "menus": {
        "editor/context": [
          {
            "command": "extension.openaiGenerate",
            "when": "editorTextFocus",
            "group": "openai"
          }
        ]
    }
```

When I run the extension I do see the new option when I select and right click. It also pops the language selection. But after choosing the language, nothing happens. There's no error message or any consol.log() 

My extension code looks like this at the moment:

```
import * as vscode from 'vscode';
import { Configuration, OpenAIApi, CreateCompletionResponse } from 'openai';

const configuration = new Configuration({
    apiKey: '***',
});

const openai = new OpenAIApi(configuration);

export function activate(context: vscode.ExtensionContext) {
    // Add a command that is triggered when the user right-clicks on a selection and selects "OpenAI Generate"
    context.subscriptions.push(vscode.commands.registerCommand('extension.openaiGenerate', () => {

      // Get the selected text
      const visibleEditors = vscode.window.visibleTextEditors;
      let editor: vscode.TextEditor | null = null;
      let selection: vscode.Selection | null = null;
      let text: string | null = null;
      
      if (visibleEditors.length > 0) {
        const editor = visibleEditors[0];
        const selection = editor.selection;
        const text = editor.document.getText(selection);
        console.log(text);
      }else{

        const editor = vscode.window.activeTextEditor;

        if (!editor) {
            vscode.window.showErrorMessage('No editor detected');
            return;
        }else{
            const selection = editor.selection;
            const text = editor.document.getText(selection);
        }
        
      }
 
      // Show a quick pick menu with the available language options
      vscode.window.showQuickPick(['python', 'javascript', 'csharp', 'java', 'go', 'ruby', 'swift', 'php', 'powershell'], {
        placeHolder: 'Select a language'
      }).then((language) => {

        if (!language) {
          return; // No language was selected
        }else{
            console.log(language);
        }
  
        // Use the OpenAI API to generate code suggestions in the selected language
        async function createCompletion(editor: vscode.TextEditor | null, selection: vscode.Selection | null, text: string | null) {
            if (editor !== null && selection !== null && text !== null){
                try {
                const response = await openai.createCompletion({
                    model: "text-davinci-002",
                    prompt: `Provide a code example in ${language} for the following description: ${text}`,
                    temperature: 0.5,
                    max_tokens: 2048,
                    top_p: 1,
                    frequency_penalty: 0,
                    presence_penalty: 0,
                    stop: ["\n"],
                });
            
                // Replace the selected text with the code suggestions
                console.log(response.data);
                editor.edit((editBuilder) => {
                    editBuilder.replace(selection, 'Hey man it works');
                });
                } catch (error) {
                console.log(error);
                }
            }
          }
          
          createCompletion(editor, selection, text);
      });
    }));
  
  }
```

Because I can't debug ... I am unsure what the response.data is returning ... so for my first test, I just wanted to replace the selected text with 'Hey man it works' ... But it's not triggered.

I think it might be an architectural error but I am not sure and still very new to TS, VSC and OpenAI for that matter.

Any help would be appreciated!

**Comment by isidorn** [comment]
@drisate I noticed you answered some discussions using AI. We'd kindly ask that you please not do that so people here know that they are talking to a person and not a bot. If this occurs again, we may remove you from the discussions and we definitely do **not** want to do that because we love all of our VS Code extension authors. We love Chat-GPT too!

As for this particular question - I suggest fix the problem that you can't debug. Every extension should be debugable https://code.visualstudio.com/api/get-started/your-first-extension#debugging-the-extension
